### PR TITLE
Set tick spacing range as constants

### PR DESCRIPTION
**Pull request by hyunchel**
## Related Issue
#371 

## Description of changes
While tick and tick spacing both use int24 as their type, each has a different range. Tick spacing has a range of [1, 32767]. This commit updates Tick test cases to use proper tick spacing range instead that of tick.

### CODE REVIEW

**Comment by hyunchel** [comment]
Updated the commit to tag the related issue.

**Comment by hyunchel** [comment]
Added and squashed missed commit.

File: test/foundry-tests/Tick.t.sol [hunk 1]
```diff
@@ -105,18 +105,18 @@ contract TickTest is Test, GasSnapshot {
...
-        checkCantOverflow(1, maxLiquidityPerTick);
+        checkCantOverflow(Constants.MIN_TICK_SPACING, maxLiquidityPerTick);
     }
 
-    function testTick_tickSpacingToMaxLiquidityPerTick_returnsTheCorrectValueForEntireRange() public {
-        uint128 maxLiquidityPerTick = tickSpacingToMaxLiquidityPerTick(887272);
+    function testTick_tickSpacingToMaxLiquidityPerTick_returnsTheCorrectValueForMaxTickSpacing() public {
+        uint128 maxLiquidityPerTick = tickSpacingToMaxLiquidityPerTick(Constants.MAX_TICK_SPACING);
 
-        assertEq(maxLiquidityPerTick, Constants.MAX_UINT128 / 3); // 126 bits
```
Comment (zhongeric) [inline test/foundry-tests/Tick.t.sol:118]: Do you mind adding the additional test you mention in #371 to this test file as well? 
```
function testTick_tickSpacingToMaxLiquidityPerTick_returnsTheCorrectValueForEntireRange() public {
    uint128 maxLiquidityPerTick = tickSpacingToMaxLiquidityPerTick(Constants.MAX_TICK);

    assertEq(maxLiquidityPerTick, type(uint128).max / 3);
    checkCantOverflow(Constants.MAX_TICK, maxLiquidityPerTick);
}
```

**Review by zhongeric — APPROVED** [review]

**Review by zhongeric — CHANGES_REQUESTED** [review]
Thanks for the contribution! Overall looks good to me, left one comment referencing something you outlined in your issue. Can you also merge in the latest main after addressing?

**Comment by hyunchel** [comment]
Thanks for the review! I've addressed your request, and have merged main into this branch as well.

**Comment by zhongeric** [comment]
Great thanks! 

On second review, I realized that we don't have `MIN/MAX_TICK_SPACING` in the TickMath library, do you mind adding them in there? I'd prefer to assign the constant to the fully resolved number with a comment saying that it's max int16. We should also probably replace all instances of `MIN/MAX_TICK` in the test file to read from TickMath instead of the test Constants file.

**Comment by hyunchel** [comment]
Sure thing! I have replaced all the references of `MIN/MAX_TICK` to use TickMath file.

There is another declaration of `MIN/MAX_TICK_SPACING` in PoolManager which is required by an interface.
I tried to minimize the change here, but if you think otherwise, please let me know!

File: contracts/PoolManager.sol [hunk 2]
```diff
@@ -34,10 +35,10 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
     using FeeLibrary for uint24;
 
     /// @inheritdoc IPoolManager
-    int24 public constant override MAX_TICK_SPACING = type(int16).max;
+    int24 public constant override MAX_TICK_SPACING = TickMath.MAX_TICK_SPACING;
```
Comment (snreynolds) [inline contracts/PoolManager.sol:38]: I know you didn't introduce this change but I think that the `override` keyword is unnecessary.

**Review by snreynolds — APPROVED** [review]
nice. good change

File: contracts/PoolManager.sol (hunk 2 above)
Comment (hyunchel) [reply contracts/PoolManager.sol]: Thanks! Removed both overrides.

**Review by zhongeric — DISMISSED** [review]
Looks good thanks for the contribution! Oops seems like there are some merge conflicts 😅 if you can fix them I'll approve again and merge for you.

**Comment by hyunchel** [comment]
Merge conflicts are resolved now. Thanks for the reviews!

**Review by zhongeric — DISMISSED** [review]

**Comment by zhongeric** [comment]
Oh shoot looks like you have to run `forge snapshot` since we are checking in the snapshots file now

**Comment by hyunchel** [comment]
Oops, sorry about that. Just pushed this PR with snapshot updates!

**Review by zhongeric — APPROVED** [review]

**Comment by zhongeric** [comment]
Yay just merged! Thanks again for the contribution :)

**Pull request merged**